quirks/password-rules.json

```diff
@@ -80,6 +80,9 @@
     "bcassessment.ca": {
         "password-rules": "minlength: 8; maxlength: 14;"
     },
+    "belkin.com": {
+        "password-rules": "minlength: 8; required: lower, upper; required: digit; required: [$!@~_,%&];"
```

Comment:
You’ve left out `-`, which needs to be the first character in a class that it’s in. So: `required: [-$!@~_,%&];`.

You cited iCloud Keychain as not being compatible. These rules seem to be compatible to me.

Reply:
Hey Ricky, 

"-" doesn't actually count towards the "Password must contain at least one of the following special characters:". This is why my iCloud generated passwords are rejected as the only special character used for mine is a "-".

I had originally assumed what looked like a "-" in the requirements pop up was actually a "~" so I replaced it in the rule. However looking back since there are other accepted characters omitted from the pop up, the inclusion of "~" is probably a mistake.

Including the "-" would break this rule in the event it is the only special character used. 

Below is a screenshot of an iCloud generated password failing. Even ignoring the requirements pop up, submitting it will reject the password.
<img width="987" alt="Screenshot 2021-11-09 at 21 06 29" src="https://user-images.githubusercontent.com/75224610/141006769-0605e65f-c131-4282-bf7c-e67a595a100e.png">

In this instance would it be preferable to include any characters tested and working in the rule, or only those suggested in the pop up that are also tested and working?

Thanks!

Reply:
Wow, tricky!